### PR TITLE
[Trino] Update drag and drop suggestions to use double quotes instead of backticks

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Modified the code for Trino editor to use double quotes (") instead of backticks (`) when dragging and dropping a table from the left assist into the editor, as Trino does not support backticks for identifiers. This change ensures that table suggestions in the editor are correctly formatted and compatible with Trino syntax.

## How was this patch tested?
Screenshot 2024-09-25 at 12 37 19 PM

- Locally

### CODE REVIEW

File: desktop/core/src/desktop/js/sql/sqlUtils.ts [hunk 1]
```diff
@@ -267,24 +267,51 @@ export default {
...
     ).getReservedKeywords(connector.dialect || 'generic');
-    if (reservedKeywords.has(identifier.toUpperCase())) {
-      return quoteChar + identifier + quoteChar;
-    }
 
-    if (!/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)) {
+    // Quote the identifier if it is a reserved keyword or not a valid SQL identifier
+    if (
+      reservedKeywords.has(identifier.toUpperCase()) ||
+      !/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/sql/sqlUtils.ts:310]: I think there are SQL dialects that allow _ as the first character as well (if we are talking about table names etc)

**Review by bjornalm — APPROVED** [review]
Looks good to me, just a small change to the Regex if I'm not mistaken.

**Review by JohanAhlen — APPROVED** [review]
LGTM!

File: desktop/core/src/desktop/js/sql/sqlUtils.ts (hunk 1 above)
Comment (agl29) [reply desktop/core/src/desktop/js/sql/sqlUtils.ts:310]: @bjornalm  Yes, I agree, but if it starts with a "-", we are enclosing the name with either backticks or double quotes.

Comment (bjornalm) [reply desktop/core/src/desktop/js/sql/sqlUtils.ts:310]: So perhaps I'm misunderstanding, but we want to "Quote the identifier if it is a reserved keyword or not a valid SQL identifier"
`reservedKeywords.has(identifier.toUpperCase())` = reserved keywords 
`!/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)` = not a valid SQL identifier
`/^[A-Za-z][A-Za-z0-9_]*$/.test(identifier)` = a valid SQL identifier

But that regex will not allow underscore as the first character of a table name, which many dialects do support.
Perhaps we should use `^[A-Za-z_][A-Za-z0-9_]*`

**Pull request merged**